src/app/job-status/job-status.component.html

```diff
@@ -30,19 +30,24 @@
     </span>
     <br />
     <div class="job-info__print-details">
-        <fa-icon [icon]="['fas', 'clock']" class="job-info__print-details-icon"></fa-icon>
-        <span
-            class="job-info__print-details-value">{{ job.estimatedPrintTime.value }}{{ job.estimatedPrintTime.unit }}</span>
-        <span class="job-info__print-details-finish-state">
-            - will finish ~{{ job.estimatedEndTime }}
+        <span *ngIf="job.hasOwnProperty('estimatedPrintTime') && job.hasOwnProperty('estimatedEndTime')">
+            <fa-icon [icon]="['fas', 'clock']" class="job-info__print-details-icon"></fa-icon>
+            <span class="job-info__print-details-value">
+                {{ job.estimatedPrintTime.value }}{{ job.estimatedPrintTime.unit }}
+            </span>
+            <span class="job-info__print-details-finish-state">
+                - will finish ~{{ job.estimatedEndTime }}
+            </span>
+            <br />
         </span>
-        <br />
-        <fa-icon [icon]="['fas', 'dharmachakra']" class="job-info__print-details-icon"></fa-icon>
-        <span class="job-info__print-details-value">
-            {{ job.filamentAmount }}g
-        </span>
-        <span class="job-info__print-details-finish-state">
-            filament will be used
+        <span *ngIf="job.hasOwnProperty('filamentAmount')">
+            <fa-icon [icon]="['fas', 'dharmachakra']" class="job-info__print-details-icon"></fa-icon>
+            <span class="job-info__print-details-value">
+                {{ job.filamentAmount }}g
+            </span>
+            <span class="job-info__print-details-finish-state">
+                filament will be used
+            </span>
```

Comment:
this will just be a big empty space if no GCode Analysis is available right?

Reply:
yes, it just hides the estimated completion time and filament weight required since they aren't available, which does leave quite a bit of empty space.